pkg/cmd/grafana-cli/commands/install_command.go

```diff
@@ -64,34 +66,56 @@ func installCommand(c utils.CommandLine) error {
 func InstallPlugin(pluginName, version string, c utils.CommandLine) error {
 	pluginFolder := c.PluginDirectory()
 	downloadURL := c.PluginURL()
+	isInternal := false
+
+	var checksum string
 	if downloadURL == "" {
-		plugin, err := s.GetPlugin(pluginName, c.RepoDirectory())
+		if strings.HasPrefix(pluginName, "grafana-") {
```

Comment:
Interesting thought but I am not familiar with these kind of things that much. Only experience I have is iOS app signing and at least the dev experience adds a some unnecessary (I think) complexity. There it is done I think mainly so that you cannot install apps from unsupported places (outside of App store) but if we want to still allow installation of non signed third party plugins not sure if the signing will have that much added value.

Reply:
We should add signing to our ecosystem for sure.  We can use that for whatever we need.  Perhaps: automatically trust anything signed by grafana and a big big warnings when installing stuff signed by other people, and even bigger warnings when not signed at all?

Reply:
Well the question is what is the end goal of that. I mean right now what you can trust is anything that comes from our plugin API I assume. If we have signed plugins I assume you could distribute it by other means and still be able to decide whether to trust it based on whether it is signed, that I assume would allow something like a third party plugin API with trusted plugins but not sure that is important for us.

I am not against that or anything I am just genuinely curious what are the benefits because from my very modest understanding of the topic I do not see that many.